pkg/storage/gcp/gcp.go

```diff
@@ -35,40 +35,94 @@ func New(ctx context.Context, gcpConf *config.GCPConfig, timeout time.Duration)
 		return nil, errors.E(op, fmt.Errorf("could not create new storage client: %s", err))
 	}
 
-	u, err := url.Parse(fmt.Sprintf("https://storage.googleapis.com/%s", gcpConf.Bucket))
-	if err != nil {
-		return nil, errors.E(op, err)
-	}
-
 	bkt := s.Bucket(gcpConf.Bucket)
 	if _, err := bkt.Attrs(ctx); err != nil {
 		if err == storage.ErrBucketNotExist {
 			return nil, errors.E(op, "You must manually create a storage bucket for Athens, see https://cloud.google.com/storage/docs/creating-buckets#storage-create-bucket-console")
 		}
-
 		return nil, errors.E(op, err)
 	}
 
 	return &Storage{
-		bucket:       &gcpBucket{bkt},
-		baseURI:      u,
+		bucket:       bkt,
 		closeStorage: s.Close,
 		timeout:      timeout,
 	}, nil
 }
 
-func newWithBucket(bkt Bucket, uri *url.URL, timeout time.Duration) *Storage {
-	return &Storage{
-		bucket:       bkt,
-		baseURI:      uri,
-		closeStorage: func() error { return nil },
-		timeout:      timeout,
-	}
-}
-
 // Close calls the underlying storage client's close method
 // It is not required to be called on program exit but provided here
 // for completness.
 func (s *Storage) Close() error {
 	return s.closeStorage()
 }
+
+func (s *Storage) delete(ctx context.Context, path string) error {
+	err := s.bucket.Object(path).Delete(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Storage) open(ctx context.Context, path string) (io.ReadCloser, error) {
+	rc, err := s.bucket.Object(path).NewReader(ctx)
+	if err != nil {
+		return rc, err
+	}
+	return rc, nil
+}
+
+func (s *Storage) write(ctx context.Context, path string) io.WriteCloser {
+	return s.bucket.Object(path).NewWriter(ctx)
+}
+
+func (s *Storage) list(ctx context.Context, prefix string) ([]string, error) {
```

Comment:
this whole function can also be inside `func (s *Storage) List` since the caller doesn't really do much otherwise.